src/process_utility.c

```diff
@@ -1430,7 +1430,9 @@ process_altertable_chunk(Hypertable *ht, Oid chunk_relid, void *arg)
 static void
 process_altertable_set_tablespace_end(Hypertable *ht, AlterTableCmd *cmd)
 {
+#if USE_ASSERT_CHECKING
```

Comment:
I think this is actually an incorrect assertion because it is valid for `get_rel_tablespace()` to return `InvalidOid` in case the table is in the default tablespace. Thus i believe it should be removed completely.

Should double-check with @Ngalstyan4, who wrote this code.